### ISSUE TITLE
as.vector does not accept arguments of type "list"

### ISSUE THREAD

**Issue by thorstenwagner** [bedatadriven/renjin#318]
This piece of code works well in R:
```
k <- apply(fprob[1:nrow(fprob),],1,which.max)
as.vector(k,mode="numeric")
```

In the latest renjin I got the following 
```
Error: (list) object cannot be coerced to type 'double'
```

The fprob object could be downloaded here:
[newfprob.zip](https://github.com/bedatadriven/renjin/files/995946/newfprob.zip)

**Comment by thorstenwagner** [comment]
Any news here or suggestions for a workaround? :-(
BTW:
as.numeric gives me the same error.

**Comment by akbertram** [comment]
Interesting case. as.vector() and friends only accept lists which contains exclusively elements which are atomic vectors of length one. For example, the expression

    as.numeric(list(1,2,3:4))

Will fail with an error in both Renjin and GNU R.

I thought we had matched GNU R's behavior exactly here, but it turns out that GNU R accepts lists of atomic vectors of length one, *or less*. The minimal example of the problem you're encountering here is:

    as.numeric(list(1,2,numeric(0)))

GNU R accepts this list, coercing the empty vector to an NA, while Renjin rejects the list.

Your example fails because `k[[1554]]` is `integer(0)`.

The following will work around the problem in Renjin:

    k <- apply(fprob[1:nrow(fprob),],1,which.max)
    as.vector(k,mode="numeric")
    k[ sapply(k, length) == 0 ] <- NA_integer_

But I will fix this properly in Renjin this week.

**Comment by thorstenwagner** [comment]
Thank you very much for your response and the workaround!

**Comment by akbertram** [comment]
This is fixed in Renjin 0.8.2364.
